pkg/controller/servicemesh/webhook/controller.go

```diff
@@ -0,0 +1,247 @@
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("webhook-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch secrets in watchedSecretNames
+	watchedSecretNames := []string{sidecarInjectorServiceAccountSecretName, galleyServiceAccountSecretName}
+	secretToPrefixMap := map[string]string{
+		sidecarInjectorServiceAccountSecretName: sidecarInjectorWebhookConfigNamePrefix,
+		galleyServiceAccountSecretName:          galleyWebhookConfigNamePrefix,
+	}
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+			requests := []reconcile.Request{}
+			if prefix := secretToPrefixMap[obj.Meta.GetName()]; prefix != "" {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      prefix + obj.Meta.GetNamespace(),
+						Namespace: "",
+					},
+				})
+			}
+			return requests
+		})}, predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			return contains(watchedSecretNames, event.Meta.GetName())
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			return contains(watchedSecretNames, event.MetaNew.GetName())
+		},
+		// deletion and generic events don't interest us
+		DeleteFunc: func(event event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(event event.GenericEvent) bool {
+			return false
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch MutatingWebhookConfigurations
+	err = c.Watch(&source.Kind{Type: &v1beta1.MutatingWebhookConfiguration{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			return strings.HasPrefix(event.Meta.GetName(), sidecarInjectorWebhookConfigNamePrefix)
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			return strings.HasPrefix(event.MetaNew.GetName(), sidecarInjectorWebhookConfigNamePrefix)
+		},
+		// deletion and generic events don't interest us
+		DeleteFunc: func(event event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(event event.GenericEvent) bool {
+			return false
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch ValidatingWebhookConfigurations
+	err = c.Watch(&source.Kind{Type: &v1beta1.ValidatingWebhookConfiguration{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			return strings.HasPrefix(event.Meta.GetName(), galleyWebhookConfigNamePrefix)
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			return strings.HasPrefix(event.MetaNew.GetName(), galleyWebhookConfigNamePrefix)
+		},
+		// deletion and generic events don't interest us
+		DeleteFunc: func(event event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(event event.GenericEvent) bool {
+			return false
+		},
+	})
```

Comment:
The duplication in these predicate functions and the ones for mutating webhooks above can be removed by using a single predicate.Funcs with a configurable prefix.